internal/document_test.go

```diff
@@ -29,16 +29,89 @@ func initDocuments(t *testing.T) (
 		attestation, err := base64.StdEncoding.DecodeString(att64String)
 		require.NoError(t, err)
 
-		cosePayload := internal.CosePayload{}
-		err = cbor.Unmarshal(attestation, &cosePayload)
+		coseSign1, err := internal.MakeCoseSign1FromBytes(attestation)
 		require.NoError(t, err)
 
-		err = cbor.Unmarshal(cosePayload.Payload, &docs[i])
+		err = cbor.Unmarshal(coseSign1.Payload, &docs[i])
 		require.NoError(t, err)
 	}
 	return docs[0], docs[1], docs[2]
 }
 
+func TestMakeDocumentFromBytes(t *testing.T) {
+	happyPathTests := []struct {
+		name           string
+		attestationB64 string
+	}{
+		{
+			"happy path - nitro",
+			internal.NitroAttestationB64,
+		},
+		{
+			"happy path - debug",
+			internal.DebugNitroAttestationB64,
+		},
+		{
+			"happy path - self signed",
+			internal.SelfSignedAttestationB64,
+		},
+	}
+
+	for _, tt := range happyPathTests {
+		t.Run(tt.name, func(t *testing.T) {
+			// given
+			attestation, err := base64.StdEncoding.DecodeString(tt.attestationB64)
+			require.NoError(t, err)
+
+			coseSign1, err := internal.MakeCoseSign1FromBytes(attestation)
+			require.NoError(t, err)
+
+			// when
+			doc, err := internal.MakeDocumentFromBytes(coseSign1.Payload)
+
+			// then
+			require.NoError(t, err)
+			require.Equal(t, doc.Digest, "SHA384")
+		})
+	}
+
+	t.Run("checking mandatory fields", func(t *testing.T) {
+		// given
+		doc := internal.Document{}
+		docBytes, err := cbor.Marshal(doc)
+		require.NoError(t, err)
+
+		// when
+		_, err = internal.MakeDocumentFromBytes(docBytes)
+
+		// then
+		assert.ErrorContains(t, err, "checking mandatory fields")
+	})
+
+	t.Run("checking optional fields", func(t *testing.T) {
+		// given
+		attestation, err := base64.StdEncoding.DecodeString(internal.NitroAttestationB64)
+		require.NoError(t, err)
+
+		coseSign1, err := internal.MakeCoseSign1FromBytes(attestation)
+		require.NoError(t, err)
+
+		doc := internal.Document{}
```

Comment:
I think it makes sense in `happy path` tests to use the `testdata` attestations. In the error tests it would be clearer to build a `doc` from scratch and then user `doc.MarshalBinary` to pass to see if it can be `UnmarshallBinary` (if that function has any checking).

Reply:
Setup is much simpler now with UnmarshalBinary